<?xml version="1.0"?>
<launch>
  <arg name="x_csv" default="$(env HOME)/Animal_Navigation/data/20200221/0.5m/x_dem_0.5m_391x409_utm.csv" />
  <arg name="y_csv" default="$(env HOME)/Animal_Navigation/data/20200221/0.5m/y_dem_0.5m_391x409_utm.csv" />
  <arg name="z_csv" default="$(env HOME)/Animal_Navigation/data/20200221/0.5m/z_dem_0.5m_391x409_utm.csv"  />
  <arg name="resolution" default="0.5" />
  <arg name="gnss_frame_id" default="gps_origin" />
  <arg name="map_frame_id" default="map" />

  <node pkg="convert_to_dem" type="text_to_dem.py" name="csv_to_dem" >
    <rosparam command="load" file="$(find convert_to_dem)/config/csv_to_dem.yaml" />

    <param name="csv/x" value="$(arg x_csv)" />
    <param name="csv/y" value="$(arg y_csv)" />
    <param name="csv/z" value="$(arg z_csv)" />
    <param name="resolution" value="$(arg resolution)" />
    <param name="gnss_frame_id" value="$(arg gnss_frame_id)" />
    <param name="map_frame_id" value="$(arg map_frame_id)" />
  </node>

  <node pkg="dem_to_occupancy" type="dem_to_occupancy" name="dem_to_occupancy" />

  <node pkg="convert_to_dem" type="dem_to_occupancy_client.py" name="dem_to_occupancy_client" >
    <param name="layer" value="traversability" />
    <param name="threshold" value="0.8" />
  </node>

  <node pkg="grid_map_demos" type="filters_demo" name="grid_map_to_traversability" output="screen">
    <param name="input_topic" value="/elevation_map" />
    <param name="output_topic" value="/filtered_map" />
    <rosparam command="load" file="$(find convert_to_dem)/config/filters.yaml" />
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find convert_to_dem)/rviz/view.rviz" />

</launch>
